pkg/apis/eventing/v1alpha1/eventtype_lifecycle.go

```diff
@@ -51,10 +56,38 @@ func (et *EventTypeStatus) MarkBrokerDoesNotExist() {
 	eventTypeCondSet.Manage(et).MarkFalse(EventTypeConditionBrokerExists, "BrokerDoesNotExist", "Broker does not exist")
 }
 
+func (et *EventTypeStatus) MarkBrokerExistsUnknown(reason, messageFormat string, messageA ...interface{}) {
+	eventTypeCondSet.Manage(et).MarkUnknown(EventTypeConditionBrokerExists, reason, messageFormat, messageA...)
+}
+
 func (et *EventTypeStatus) MarkBrokerReady() {
 	eventTypeCondSet.Manage(et).MarkTrue(EventTypeConditionBrokerReady)
 }
 
-func (et *EventTypeStatus) MarkBrokerNotReady() {
-	eventTypeCondSet.Manage(et).MarkFalse(EventTypeConditionBrokerReady, "BrokerNotReady", "Broker is not ready")
+func (et *EventTypeStatus) MarkBrokerFailed(reason, messageFormat string, messageA ...interface{}) {
+	eventTypeCondSet.Manage(et).MarkFalse(EventTypeConditionBrokerReady, reason, messageFormat, messageA...)
+}
+
+func (et *EventTypeStatus) MarkBrokerUnknown(reason, messageFormat string, messageA ...interface{}) {
+	eventTypeCondSet.Manage(et).MarkUnknown(EventTypeConditionBrokerReady, reason, messageFormat, messageA...)
+}
+
+func (et *EventTypeStatus) PropagateBrokerStatus(bs *BrokerStatus) {
+	bc := brokerCondSet.Manage(bs).GetTopLevelCondition()
+	if bc == nil {
+		et.MarkBrokerUnknown("BrokerUnknown", "The condition of Broker is nil")
+		return
+	}
+	if bc.IsTrue() {
+		eventTypeCondSet.Manage(et).MarkTrue(EventTypeConditionBrokerReady)
+	} else {
+		msg := bc.Message
+		if bc.IsUnknown() {
+			et.MarkBrokerUnknown("BrokerUnknown", "The status of Broker is Unknown: %s", msg)
```

Comment:
I think you can just use the condition directly to propagate:  https://github.com/knative/serving/blob/ec457adcb1466abfcbc1c0a069e60285e769d135/pkg/apis/serving/v1alpha1/route_lifecycle.go#L157-L171

The current way is dropping the real reason and messages so it makes it hard to use.